components/notebook-controller/controllers/notebook_controller.go

```diff
@@ -52,13 +53,48 @@ const DefaultServingPort = 80
 const AnnotationRewriteURI = "notebooks.kubeflow.org/http-rewrite-uri"
 const AnnotationHeadersRequestSet = "notebooks.kubeflow.org/http-headers-request-set"
 const AnnotationNotebookRestart = "notebooks.opendatahub.io/notebook-restart"
+// annotation that makes OpenShift ImagePolicy admission plug-in resolve the image-field from an imagestream name:tag reference
+const AnnotationImageChangeTrigger = "image.openshift.io/triggers"
 
 const PrefixEnvVar = "NB_PREFIX"
 
 // The default fsGroup of PodSecurityContext.
 // https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.11/#podsecuritycontext-v1-core
 const DefaultFSGroup = int64(100)
 
+func getContainerNamesFromAnnotationImageChangeFieldPaths(dataJson string) []string {
+	annotationImageChangeContent := AnnotationImageChangeContent{}
+	err := json.Unmarshal([]byte(dataJson), &annotationImageChangeContent)
+	if err != nil {
+		fmt.Println("JSON decode error!")
```

Comment:
@harshad16 I don't see fmt fmt-Println anywhere else. Now, this json from the annotation decode error is very unlikely to occur, but still, it should be output correctly. Better use

		`log.Error("JSON decode error: %v ", err)`      ?
		
I am using log.info further down below to log to kubeflow notebook controller stdout which container's image field value is excluded from reconciliation.